src/coreclr/src/interop/comwrappers.hpp

```diff
@@ -139,7 +135,9 @@ class NativeObjectWrapperContext
 {
     IReferenceTracker* _trackerObject;
     void* _runtimeContext;
-    Volatile<BOOL> _isValidTracker;
+    Volatile<BOOL> _trackerObjectDisconnected;
```

Comment:
Any particular reason we use `BOOL` instead of `bool` here?

Reply:
Sort of. I had assumed the `Volatile<T>` type didn't support `bool`, but I just looked and there are several places where it is used. However the real underlying issue is that this is used in a call to `InterlockedCompareExchange` which assumes at least a 4 byte volatile memory location. I don't want to play with any of the C++ threading primitives here since it does tend to bloat the binary size when linking in the concrt for Windows. Open to other options though.